README.rst

```diff
@@ -1,109 +1,138 @@
 
-<sub>Figure from Jordan, Schmidt, Senn & Petrovici, "Evolving to learn: discovering interpretable plasticity rules for spiking networks", [ arxiv:2005.14149](https://arxiv.org/abs/2005.14149).</sub>
+.. _arxiv:2005.14149: https://arxiv.org/abs/2005.14149
 
+.. long-description-end
 
 A simple example of CGP applied to a symbolic regression problem can be found in `examples/example_evo_regression.py`.
 
-Searching numerical values for constants via local search
----------------------------------------------------------
-
-This library supports the use of constants as operator nodes in order to represent expressions like `f(x) = x + c` with some fixed value `c`.
-While the `ConstantFloat` node and derived nodes have a fixed output value, the output value of `Parameter` nodes are stored per individual and hence can be modified during evolution via local search.
-The library provides a local search function that performs stochastic gradient descent on the values of `Parameter` nodes (cf. Topchy & Punch, 2001; Izzo et al., 2017).
-See `examples/example_differential_evo_regression.py` for an example evolving an expression containing mathematical constants.
-
+.. installation-start
+============
 Installation
 ============
 
 You can install the latest relase via pip:
-```shell script
-pip install hal-cgp
-```
+
+   .. code-block:: shell
+
+      pip install hal-cgp
+
 
 This library depends on some optional packages defined in `extra_requirements.txt`. These are necessary, for example, to compile an individual to a SymPy expression or a PyTorch class. You can install all extra requirements via:
-```shell script
-pip install hal-cgp[all]
-```
+
+   .. code-block:: shell
+
+      pip install hal-cgp[all]
 
 You can also install individual extra requirements by specifying the package name (without version number) in square brackets, e.g., to install the `torch` dependency:
-```shell script
-pip install hal-cgp[torch]
-```
+
+   .. code-block:: shell
+
+      pip install hal-cgp[torch]
 
 The adventurous can install the most recent development version directly from our master branch (don't use this in production unless there are good reasons!):
-```shell script
-git clone git@github.com:Happy-Algorithms-League/hal-cgp.git
-cd hal-cgp
-pip install .[all]
-```
 
+   .. code-block:: shell
+
+      git clone git@github.com:Happy-Algorithms-League/hal-cgp.git
+      cd hal-cgp
+      pip install .[all]
 
+.. installation-end
+.. basic-usage-start
+===========
 Basic usage
 ===========
 
 Follow these steps to solve a basic regression problem:
 
 1. Define an objective function.
    The objective function takes an individual as an argument and updates the `fitness` of the individual.
-```python
-def objective(individual):
-    individual.fitness = ...
-    return individual
-```
+
+   .. code-block:: python
+		
+      def objective(individual):
+          individual.fitness = ...
```

Comment:
this looks like too much indentation

Reply:
I think this looks fine in the rendered README because it aligns with the enumerated list items, please check here: https://github.com/Happy-Algorithms-League/hal-cgp/tree/doc/sphinx_docs

same for all the other comments below.